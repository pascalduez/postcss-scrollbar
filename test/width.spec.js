import postcss from 'postcss';
import { stripIndent } from 'common-tags';
import plugin from '../src';

let from, to;

describe('width:', () => {
  test('auto keyword', () => {
    let input = stripIndent`
      .test {
        scrollbar-width: auto;
      }
    `;

    return postcss([plugin()])
      .process(input, { from, to })
      .then(result => {
        expect(result.css).toMatchSnapshot();
      });
  });

  test('thin keyword', () => {
    let input = stripIndent`
      .test {
        scrollbar-width: thin;
      }
    `;

    return postcss([plugin()])
      .process(input, { from, to })
      .then(result => {
        expect(result.css).toMatchSnapshot();
      });
  });

  test('none keyword', () => {
    let input = stripIndent`
      .test {
        scrollbar-width: none;
      }
    `;

    return postcss([plugin()])
      .process(input, { from, to })
      .then(result => {
        expect(result.css).toMatchSnapshot();
      });
  });

  test('erroneous keyword', async () => {
    let input = stripIndent`
      .test {
        scrollbar-width: 1rem;
      }
    `;

    let result = await postcss().use(plugin).process(input, { from, to });

    expect(result.css).toMatchSnapshot();
    expect(result.messages.length).toBeGreaterThan(0);
    expect(result.messages[0].type).toBe('warning');
    expect(result.messages[0].text).toMatch(
      /Invalid value for property `scrollbar-width`/
    );
  });
});
