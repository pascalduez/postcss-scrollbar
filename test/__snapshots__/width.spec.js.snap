// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`width: auto keyword 1`] = `
".test::-webkit-scrollbar {
  width: initial;
  height: initial;
}
.test {
  -ms-overflow-style: auto;
  scrollbar-width: auto;
}"
`;

exports[`width: erroneous keyword 1`] = `
".test {
  scrollbar-width: 1rem;
}"
`;

exports[`width: none keyword 1`] = `
".test::-webkit-scrollbar {
  width: 0;
  height: 0;
}
.test {
  -ms-overflow-style: none;
  scrollbar-width: none;
}"
`;

exports[`width: thin keyword 1`] = `
".test::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}
.test {
  -ms-overflow-style: auto;
  scrollbar-width: thin;
}"
`;
